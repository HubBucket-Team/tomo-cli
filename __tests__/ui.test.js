import React from 'react';
import {render} from 'ink-testing-library';
import Tomo, {Warning, Task, TaskList} from '../src/ui';

const ARROW_DOWN = '\u001B[B';

describe('Task component', () => {
    test('can render', () => {
        const text = 'test task text';
        const component = render(<Task text={text}></Task>);
        expect(component.lastFrame()).toMatchSnapshot();
    });
    test('can render (completed)', () => {
        const text = 'test task text';
        const {lastFrame} = render(<Task text={text} isComplete={true}></Task>);
        expect(lastFrame()).toMatchSnapshot();
    });
    test('can render with defaults', () => {
        const {lastFrame} = render(<Task isComplete={true}></Task>);
        expect(lastFrame()).toMatchSnapshot();
    });
});
describe('TaskList component', () => {
    test('can render', () => {
        const options = {skipInstall: true};
        const {lastFrame} = render(<TaskList command={'add'} terms={['eslint']} options={options}></TaskList>);
        expect(lastFrame()).toMatchSnapshot();
    });
});
describe('Warning', () => {
    test('can render', () => {
        const callback = jest.fn();
        const {lastFrame} = render(<Warning callback={callback}>Hello World</Warning>);
        expect(lastFrame()).toMatchSnapshot();
    });
});
describe('tomo', () => {
    test('add', () => {
        const input = ['add'];
        const {lastFrame, stdin} = render(<Tomo input={input}/>);
        expect(lastFrame()).toMatchSnapshot();
        stdin.write(ARROW_DOWN);
        expect(lastFrame()).toMatchSnapshot();
    });
    test('add eslint', () => {
        const input = ['add', 'eslint'];
        const {lastFrame} = render(<Tomo input={input}/>);
        expect(lastFrame()).toMatchSnapshot();
    });
    test('add eslint --use-react', () => {
        const input = ['add', 'eslint'];
        const flags = {useReact: true};
        const {lastFrame} = render(<Tomo input={input} flags={flags}/>);
        expect(lastFrame()).toMatchSnapshot();
    });
});
