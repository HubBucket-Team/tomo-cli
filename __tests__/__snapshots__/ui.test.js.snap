// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task component can render (completed) 1`] = `"   [1m[32m✔[39m[22m [2mtest task text[22m"`;

exports[`Task component can render 1`] = `"   [36m⠋[39m[39m [39m[39mtest task text[39m"`;

exports[`Task component can render with defaults 1`] = `"   [1m[32m✔[39m[22m [2mtask description[22m"`;

exports[`TaskList component can render 1`] = `
"[36m[39m
[36m ╭────────────╮[39m
 [36m│[39m [37m[1madd eslint[22m[39m [36m│[39m
[36m ╰────────────╯[39m
   [36m⠋[39m[39m [39m[39mCreate Babel config file[39m
   [36m⠋[39m[39m [39m[39mInstall Babel dependencies[39m
   [36m⠋[39m[39m [39m[39mInstall Babel presets[39m
   [36m⠋[39m[39m [39m[39mInstall Babel plugins[39m
   [36m⠋[39m[39m [39m[39mCreate ESLint config file[39m
   [36m⠋[39m[39m [39m[39mAdd lint tasks to package.json[39m
   [36m⠋[39m[39m [39m[39mInstall ESLint dependencies[39m

"
`;

exports[`Warning can render 1`] = `
"[33m[39m
[33m ┌─────────┐[39m
 [33m│[39m [33moops...[39m [33m│[39m
[33m └─────────┘[39m
    ↳ Hello World

      [2mPress [22m[1mENTER[22m[2m to continue[22m
"
`;

exports[`tomo add 1`] = `
"
 [36m[1m→[22m[39m [39m[36m[1ma11y[22m[39m[39m
   babel
   docs
   eslint
   jest
   postcss
   webpack
"
`;

exports[`tomo add 2`] = `
"
   a11y
 [36m[1m→[22m[39m [39m[36m[1mbabel[22m[39m[39m
   docs
   eslint
   jest
   postcss
   webpack
"
`;

exports[`tomo add eslint --use-react 1`] = `
"[36m[39m
[36m ╭────────────╮[39m
 [36m│[39m [37m[1madd eslint[22m[39m [36m│[39m
[36m ╰────────────╯[39m
   [36m⠋[39m[39m [39m[39mCreate Babel config file[39m
   [36m⠋[39m[39m [39m[39mInstall Babel dependencies[39m
   [36m⠋[39m[39m [39m[39mInstall Babel presets[39m
   [36m⠋[39m[39m [39m[39mInstall Babel plugins[39m
   [36m⠋[39m[39m [39m[39mInstall Babel React preset[39m
   [36m⠋[39m[39m [39m[39mAdd React support to Babel configuration file[39m
   [36m⠋[39m[39m [39m[39mCreate ESLint config file[39m
   [36m⠋[39m[39m [39m[39mAdd lint tasks to package.json[39m
   [36m⠋[39m[39m [39m[39mInstall ESLint dependencies[39m
   [36m⠋[39m[39m [39m[39mInstall ESLint React plugins[39m
   [36m⠋[39m[39m [39m[39mAdd React support to ESLint configuration file[39m

"
`;

exports[`tomo add eslint 1`] = `
"[36m[39m
[36m ╭────────────╮[39m
 [36m│[39m [37m[1madd eslint[22m[39m [36m│[39m
[36m ╰────────────╯[39m
   [36m⠋[39m[39m [39m[39mCreate Babel config file[39m
   [36m⠋[39m[39m [39m[39mInstall Babel dependencies[39m
   [36m⠋[39m[39m [39m[39mInstall Babel presets[39m
   [36m⠋[39m[39m [39m[39mInstall Babel plugins[39m
   [36m⠋[39m[39m [39m[39mCreate ESLint config file[39m
   [36m⠋[39m[39m [39m[39mAdd lint tasks to package.json[39m
   [36m⠋[39m[39m [39m[39mInstall ESLint dependencies[39m

"
`;
